projects/igniteui-angular/migrations/common/tsUtils.ts

```diff
@@ -223,23 +223,27 @@ export function createProjectService(serverHost: tss.server.ServerHost): tss.ser
 export function getTypeDefinitionAtPosition(langServ: tss.LanguageService, entryPath: string, position: number): tss.DefinitionInfo | null {
     const definition = langServ.getDefinitionAndBoundSpan(entryPath, position)?.definitions[0];
     if (!definition) { return null; }
+    // if the definition's kind is a reference, the identifier is a template variable referred in an internal/external template
     if (definition.kind.toString() === 'reference') {
-        // template variable in an internal/external template
         return langServ.getDefinitionAndBoundSpan(entryPath, definition.textSpan.start).definitions[0];
     }
     let typeDefs = langServ.getTypeDefinitionAtPosition(entryPath, definition.textSpan.start);
+    // if there are no type definitions found, the identifier is a ts property, referred in an internal/external template
     if (!typeDefs) {
-        // ts property referred in an internal/external template
-        const classDeclaration = langServ
-            .getProgram()
-            .getSourceFile(definition.fileName)
+        const sourceFile = langServ.getProgram().getSourceFile(definition.fileName);
+        // if the language service cannot resolve the source, this means its not a .ts file
```

Comment:
Actually, it seems the service can very well be tricked to try to resolve html files and boy it produces some funky results when `<custom-elem class="some-class">` can trick it to produce actual but very broken `ClassDeclaration` that has members without names. Might be wise to filter the file type ourselves TBH.